File: Array/Max Consecutive Ones/solution1.py
from typing import List


class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        max_con = count = 0
        for num in nums:
            if num == 0:
                max_con = max(max_con, count)
                count = 0
            else:
                count += 1
        max_con = max(max_con, count)
        return max_con


s = Solution()
print(s.findMaxConsecutiveOnes([1, 1, 0, 1, 1, 1]))

File: Array/Longest Consecutive Sequence/solution1.go
package main

import (
	"fmt"
)

func longestConsecutive(nums []int) int {
	m := make(map[int]bool)
	for _, num := range nums {
		m[num] = true
	}
	longest := 0
	count := 0
	for num := range m {
		if _, ok := m[num-1]; !ok {
			start := num
			for {
				if _, ok := m[start]; ok {
					start++
					count++
				} else {
					longest = max(longest, count)
					count = 0
					break
				}

			}
		}
	}
	return longest
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}
func main() {
	nums := []int{100, 4, 200, 1, 3, 2}
	// nums := []int{0, -1}
	fmt.Println(longestConsecutive(nums))
}
